imapautofiler/client.py

```diff
@@ -113,15 +114,23 @@ def expunge(self):
     def close(self):
         "Close the connection, flushing any pending changes."
 
-
+'/Users/olsenmo/Desktop/internal.pem'
 class IMAPClient(Client):
 
     def __init__(self, cfg):
         super().__init__(cfg)
+
+        # Use default client behavior if ca_file not provided.
+        context = None  
+        if 'ca_file' in cfg['server']:
+            context = ssl.create_default_context(cafile=cfg['server']['ca_file'])
+
         self._conn = imapclient.IMAPClient(
             cfg['server']['hostname'],
             use_uid=True,
             ssl=True,
+            port=cfg['server'].get('port'),
```

Comment:
Is passing None explicitly the same as the default for port? Or should we set the default explicitly in the get() call so we always pass a valid number?

Reply:
The reason I used None instead of the default was that it allows the library we are calling the ability to change the default value (say if it became a different type for example). Also the default for this value varies depending on whether ssl is True or not. If we hand the "default" value then any changes that would make ssl optional in the future would need to be aware of this.

Reply:
OK. I'm asking about what the underlying library actually expects or supports. I don't have the library code handy to look at what it's doing with a value of None.

Reply:
Ah, that makes sense. Here is what the init definition looks like:

    def __init__(self, host, port=None, use_uid=True, ssl=True, stream=False,
                 ssl_context=None, timeout=None):

If port is None it will be set based on what the value of ssl is set to.

Reply:
OK, it sounds like None is fine as a default then. Thanks for verifying!

Reply:
No problem